File: fix_images.js
const fs = require('fs');

function fixImages(filePath) {
    let html = fs.readFileSync(filePath, 'utf8');

    // Quick replace for any img containing 'women' or 'men' product images that lacks width
    let pattern = /<img[^>]*src="[^"]*images\/products\/(women|men)\/\d+\.jpg"[^>]*>/g;

    html = html.replace(pattern, (match) => {
        if (match.includes('width=')) return match;

        let newImg = match.replace('>', ' width="400" height="600" loading="lazy">');

        // Also ensure it has class="product-image" so we know the lazy class exists if needed.
        if (!newImg.includes('class=')) {
            newImg = newImg.replace('>', ' class="product-image">');
        } else if (!newImg.includes('product-image') && match.includes('class=')) {
            newImg = newImg.replace(/class="/, 'class="product-image ');
        }

        return newImg;
    });

    fs.writeFileSync(filePath, html);
    console.log('Fixed images in ' + filePath);
}

fixImages('pages/women.html');
fixImages('pages/men.html');

File: fix_images.py
import os
import re

for f in ['pages/women.html', 'pages/men.html']:
    with open(f, 'r', encoding='utf-8') as file:
        content = file.read()
        
    def replace_img(match):
        img_tag = match.group(0)
        if 'width="400"' in img_tag:
            return img_tag
        return img_tag.replace('>', ' width="400" height="600" loading="lazy">')
        
    new_content = re.sub(r'<img\s+src="(?:\.\.)?/images/products/(?:women|men|kids)/\d+\.jpg"[^>]*>', replace_img, content)
    
    with open(f, 'w', encoding='utf-8') as file:
        file.write(new_content)

print('Images fixed in women.html and men.html.')
